File: frontend/src/lib/models/workspace.ts
import type { Channel } from './channel';
import type { User } from './user';

export interface ImagePathObject {
	String: string;
	Valid: boolean;
}

export type Workspace = {
	Id: string;
	ImagePath: ImagePathObject;
	Name: string;
	Users: User[];
	Channels: Channel[];
	Permissions: string[];
};

File: backend/internal/models/workspace.go
package models

import (
	"database/sql"

	"github.com/google/uuid"
)

type Workspace struct {
	Id        uuid.UUID
	ImagePath sql.NullString
	Name      string
}

type WorkspaceFullData struct {
	Id        uuid.UUID
	ImagePath sql.NullString
	Name      string
	Users     []User
}